consensus/src/consensus/consensus_proxy.rs

```diff
@@ -352,9 +352,25 @@ impl<N: Network> ConsensusProxy<N> {
                         if response.block.block_number() <= election_head.block_number() {
                             let block_hash = response.block.hash();
                             let mut already_proven = false;
+
+                            if response.block.block_number() == Policy::genesis_block_number() {
+                                let genesis_hash = self
+                                    .blockchain
+                                    .read()
+                                    .get_block_at(Policy::genesis_block_number(), false)
+                                    .unwrap()
+                                    .hash();
+
+                                if genesis_hash == response.block.hash() {
+                                    already_proven = true;
+                                } else {
+                                    log::warn!(peer = %peer_id, "The genesis hash from the peer does not match our own");
```

Comment:
I believe a connection between peers with different genesis blocks can't even get past the initial handshake phase, so this situation _should_ be impossible.

Reply:
And the idea here is then in this scenario, this continues but then fails validation somewhere else?

Could you add a test for this malicious behavior to make sure this actually correctly fails somewhere then?